neonvm/hack/kernel/Dockerfile.kernel-builder

```diff
@@ -46,16 +48,31 @@ RUN set -e \
     && tar --strip-components=1 -C linux-${KERNEL_VERSION} -xf linux-${KERNEL_VERSION}.tar.xz
 
 
-FROM build-deps AS build
 
-ARG KERNEL_VERSION
+### Cross-compilation related steps
 
+# Build the kernel on amd64
+FROM build-deps AS build_amd64
+ARG KERNEL_VERSION
 ADD linux-config-${KERNEL_VERSION} linux-${KERNEL_VERSION}/.config
+RUN cd linux-${KERNEL_VERSION} && make ARCH=x86_64 CROSS_COMPILE=x86_64-linux-gnu- -j `nproc`
+
+# Copy the kernel image to a separate step
+# Use alpine so that `cp` is available when loading custom kernels for the runner pod.
+# See the neonvm controller's pod creation logic for more detail.
+FROM alpine:3.19 AS kernel_amd64
+ARG KERNEL_VERSION
+COPY --from=build_amd64 /build/linux-${KERNEL_VERSION}/arch/x86/boot/bzImage /vmlinuz
 
-RUN cd linux-${KERNEL_VERSION} && make -j `nproc`
+# Build the kernel on amd64
+FROM build-deps AS build_arm64
+ARG KERNEL_VERSION
+ADD linux-config-${KERNEL_VERSION} linux-${KERNEL_VERSION}/.config
+RUN cd linux-${KERNEL_VERSION} && make ARCH=arm64 CROSS_COMPILE=aarch64-linux-gnu- -j `nproc`
 
+# Copy the kernel image to a separate step
 # Use alpine so that `cp` is available when loading custom kernels for the runner pod.
 # See the neonvm controller's pod creation logic for more detail.
-FROM alpine:3.19
+FROM alpine:3.18 AS kernel_arm64
```

Comment:
A question for you: This `alpine` image is executed as-is on the host where we run the VM. Do we need to do anything special to make sure that we get the arm64 versions of packages?

Reply:
It doesn't really matter at this step which architecture we run on, the compilation is going through cross-compiling toolchain and relies on the specific compiler packages (gcc-x86-64-linux-gnu and gcc-aarch64-linux-gnu)

Reply:
What I mean is that the `cp` from that alpine image can be used directly:

https://github.com/neondatabase/autoscaling/blob/351b7b4d8c4ff13e5c97d5ae4a83ccc05efd5c26/neonvm/controllers/vm_controller.go#L1702

It's ok if this PR doesn't address that, but I want to make sure that's not forgotten :)

Reply:
Got you, looking into that 👀